java/compaction-core/src/main/java/sleeper/compaction/job/CompactionJobFactory.java

```diff
@@ -91,8 +91,28 @@ public CompactionJob createSplittingCompactionJob(
         return compactionJob;
     }
 
+    public CompactionJob createCompactionJobWithTask(
+            List<FileInfo> files, String partition, String taskId) {
```

Comment:
A compaction job will never get a task ID at the time it's created. It gets assigned when it's pulled off the queue.